docs: add --no-exec param in 'import' command (#1900)
iterative/dvc.org: content/docs/command-reference/import.md

```usage
usage: dvc import [-h] [-q | -v]
                  [-o <path>] [--file <filename>] [--rev <commit>] [--no-exec]

>                   url path

Reviewer: That new line is too long. Let's reformat this way please:
```
usage: dvc import [-h] [-q | -v] [-o <path>] [--file <filename>]
                  [--rev <commit>] [--no-exec]
                  url path
```

Author: I've moved it to a new line.

  > [Importing and updating fixed revisions](#example-importing-and-updating-fixed-revisions)
  > example below).

- `--no-exec` - create `.dvc` file without actually downloading `url`. E.g. if
  the file or directory already exist it can be used to skip download.

>   `dvc commit <out>.dvc` should be used to calculate the URL and data hash,

Reviewer 2: @BurnzZ not sure about the `calculate URL` part, could you clarify what's happening here?

Reviewer: It's a copy/paste of the homologue option in https://dvc.org/doc/command-reference/import-url#options

Reviewer: I.e. yes, that part about calculating hashes doesn't apply to DVC repo imports, we can just remove it as no other similar process occurs I think.

Reviewer 3: It should be `existS` and probably `THE download` or `downloadING`.

`E.g. if the file or directory already exist it can be used to skip download.` I'm not sure I understand this scenario? 😄 Do we run `dvc import --no-exec` on top of an existing directory - what is the goal? I thought, the primary use case is to postpone the download because we need a partial run (that is supported in `pull` but not `import`) OR the download needs to be run in another machine (we just cook metadata).

Reviewer 2: @jorgeorpinel to my mind `--no-exec` in the import-url has to be updated as well. `Calculate URL and data hash` part is not clear to me.

@dmpetrov it is the same as `import-url --no-exec`. Scenario is - create .dvc but don't download (if data exists already). E.g. you did `wget something` and now want to have a .dvc files to track it, but don't want to download data again.

Reviewer 2: `the download needs to be run in another machine (we just cook metadata).` - this scenario fits as well, I guess

`we just cook metadata` - is a good description I think.

Author: Hi everyone! Thanks for the feedback!

> It's a copy/paste of the same option in https://dvc.org/doc/command-reference/import-url#options.
@jorgeorpinel That's right! I've also updated the docs for `import-url` as per @shcheklein's reminder.

I've refined the `--no-exec` definition as per your feedback.

> - `--no-exec` - create `.dvc` file without actually downloading `url`. E.g. if
>   the file or directory already exist it can be used to skip download.
>   `dvc commit <out>.dvc` should be used to calculate the URL and data hash,
>   update the `.dvc` files, and save existing data to the cache.

Reviewer: ```suggestion
- `--no-exec` - create a `.dvc` file without actually downloading `url`. E.g. if
  the file or directory already exist, this option can be used to skip the download.
  `dvc commit <out>.dvc` should be used to update the import `.dvc` files, and save existing data to the cache.
```

Reviewer: BTW we should apply similar copy edits to `import-url --no-exec` — I'll do it separately 🙂

Reviewer 2: Btw, it should be not `url` in this case I think. `W/o downloading data or file` or something like that.

Author: Updated.

  > example below).

> - `--no-exec` - create `.dvc` file without actually downloading `url`. E.g. if

Reviewer 2: I think `url` is not the best term here (mind the difference in the command names - import vs import-_url_)

Author: Right, that makes sense since `dvc import` handles repo locations. Would `repo url` fit much better here?

Moreover, if we do replace the `url` with something else, we'll also need to update the CLI parser (_like the screenshot below_) as well as the rest of the doc for consistency.

![image](https://user-images.githubusercontent.com/3449761/97841416-bf441080-1d20-11eb-80d3-56b7377de682.png)

  the file or directory already exists, this can be used to skip the download.

>   The data hash is not calculated by this since only the metadata is saved into

Reviewer 2: @efiop when we import objects do we trust the external source for the hash? if not, does it makes sense to do this?  It seems to me that `--no-exec` at least in the case of `import` can write hash as well if the goal is to avoid data download. WDYT?

Reviewer 4: It is more complicated than that, but yeah, we trust it indirectly by trusting the cache. This is because we don't really `download`, but rather `pull + checkout`. We don't fill it on `--no-exec` for the `run` symmetry, but yeah, could consider it, though that has itsown implications. Let's leave it as is for now, since it is not a high priority.

Reviewer 2: >  but rather pull + checkout

hmm, how do we do `pull` if hash is not set? I thought that we don't do anything with data at all - kinda create .dvc with some meta in it and that it.

if we do `pull` + `checkout` what happens if data already exists in workspace and is in cache, or exists and not in cache?

> Let's leave it as is for now, since it is not a high priority. 

PR is merged, yep.

Reviewer: Pretty late to this conversation but just wanted to double check that with --no-exec there's no connection to the data source, right? Because I'm reviewing this text for an upcoming docs update. Thanks
